packages/core/src/core.ts

```diff
@@ -72,10 +76,19 @@ export async function prove(
   const anonAadhaarClaim: AnonAadhaarClaim = {
     pubKey: args.pubKey.value,
     signalHash: args.signalHash.value,
-    ageAbove18: anonAadhaarProof.ageAbove18 === '1',
-    gender: anonAadhaarProof.gender,
-    pincode: anonAadhaarProof.pincode,
-    state: convertRevealBigIntToString(anonAadhaarProof.state),
+    ageAbove18:
+      args.revealAgeAbove18.value === '1'
+        ? anonAadhaarProof.ageAbove18 === '1'
+        : null,
+    gender:
+      convertRevealBigIntToString(anonAadhaarProof.gender) === ''
```

Comment:
`convertRevealBigIntToString(anonAadhaarProof.gender) || null` would do the same right? For others as well.

Reply:
No, as the output will be an empty string but `anonAadhaarProof.gender || null` should work.

Reply:
If left side is empty string `||` will take the value on the right - so I think its the same.

Reply:
Oooh!
Let me try

Reply:
It works 
Thanks :)